packages/backend/src/graphql/queries/get-flow-transfer-details.ts

```diff
@@ -0,0 +1,63 @@
+import { ITransferDetails } from '@plumber/types'
+
+import { getEmptyConnectionDetails } from '@/helpers/get-basic-transfer-details'
+import globalVariable from '@/helpers/global-variable'
+import Context from '@/types/express/context'
+
+type Params = {
+  flowId: string
+}
+
+const getFlowTransferDetails = async (
+  _parent: unknown,
+  params: Params,
+  context: Context,
+) => {
+  const flowTransferDetails: ITransferDetails[] = []
+
+  const flow = await context.currentUser
+    .$relatedQuery('flows')
+    .findById(params.flowId)
+    .withGraphFetched({
+      steps: {
+        connection: true,
+      },
+    })
+    .throwIfNotFound('Pipe not found')
+
+  const getTransferDetailsPromises = flow.steps.map(async (step) => {
+    const app = await step.getApp()
+    // no app is selected or app has no transfer details
+    if (!app || !app.getTransferDetails) {
```

Comment:
nit: it looks like you defined a default `getTransferDetails` for all current apps with a connection. Thinking we can omit the need for that - would it be better to do this instead:

1.  check if app requires connections here (i.e. `auth` is nonnull)
2. If app needs connections and `getTransferDetails` is null, invoke the basic get transfer details.

LMK what you think, don't need to implement. If you are OK, I can help implement for you also.

Reply:
as discussed, this is a good idea cus future apps with connections will just have the default transfer details without the need for defining, thanks!